packages/neos-ui-editors/src/SecondaryEditors/MediaSelectionScreen/index.js

```diff
@@ -24,13 +24,33 @@ class MediaSelectionScreen extends PureComponent {
                 onComplete(assetIdentifier);
             }
         };
-
+        let {constraints} = this.props;
+        // Add media type constraint if (deprecated) "type" prop is set and media type constraint is not explicitly set already
+        if (type === 'images' && !constraints.mediaTypes) {
+            constraints = {...constraints, mediaTypes: ['image/*']};
+        }
         const mediaBrowserUri = $get('routes.core.modules.mediaBrowser', neos);
-
         return (
-            <iframe name="neos-media-selection-screen" src={`${mediaBrowserUri}/${type}.html`} className={style.iframe}/>
+            <iframe name="neos-media-selection-screen" src={`${mediaBrowserUri}/assets/index.html?${this.encodeAsQueryString({constraints})}`} className={style.iframe}/>
         );
     }
+
+    encodeAsQueryString = (obj, prefix) => {
```

Comment:
Maybe we put this into the `utils-helpers` package. Seems general and can be used also in other places. What do you think @bwaidelich ??

Reply:
Good call. There is already a similar helper in https://github.com/neos/neos-ui/blob/master/packages/neos-ui-backend-connector/src/Endpoints/Helpers.ts#L42

But I wasn't sure wether we want to introduce that cross-package dependency here. Also the existing helper doesn't (yet) work recursively yet IIRC and doesn't remove `null` params.

If that dependency is not an issue we could just extend the existing helper and use that

Reply:
@markusguenther @Sebobo @dimaip do you guys think its fine to have a dependency from the `neos-ui-editors` package to the `neos-ui-backend-connector` package (or is that the case already?) and – if so – should we extend the existing `urlWithParams` helper so that it can deal with nested objects and `null` params?

Reply:
*ping* :)

Reply:
Hey @bwaidelich!
I'd rather extract this helper from both places and put it into `utils-helpers`.
But I don't think it's the end of the world if you leave it here either, though extracting it is perhaps a better move.

Reply:
@markusguenther @dimaip Thanks for your feedback, I'll try to move it into `utils-helper` later today, seems like a good place for it! :)

Reply:
OK, later _today_ didn't work out but almost :)